pyqasm/modules/base.py

```diff
@@ -474,7 +474,7 @@ def unroll(self, **kwargs):
             )
             raise err
 
-    def dumps(self) -> str:
+    def qasm_str(self) -> str:
```

Comment:
should we just make this `def __str__(self)` ? i.e. so that `str(module)` just returns the OpenQASM string? Sort of like how `str(kernel)` returns the raw Quake string in `cudaq`:

```python
import cudaq

kernel = cudaq.make_kernel()

qubit = kernel.qalloc()

kernel.h(qubit)
kernel.x(qubit)
kernel.y(qubit)
kernel.z(qubit)
kernel.t(qubit)
kernel.s(qubit)

kernel.mz(qubit)

print(type(kernel))
print()
print(repr(kernel))
print()
print(str(kernel))
```
```
<class 'cudaq.kernel.kernel_builder.PyKernel'>

<cudaq.kernel.kernel_builder.PyKernel object at 0x7fbb2ed06110>

module attributes {quake.mangled_name_map = {__nvqpp__mlirgen____nvqppBuilderKernel_21TNV5JK3V = "__nvqpp__mlirgen____nvqppBuilderKernel_21TNV5JK3V_PyKernelEntryPointRewrite"}} {
  func.func @__nvqpp__mlirgen____nvqppBuilderKernel_21TNV5JK3V() attributes {"cudaq-entrypoint"} {
    %0 = quake.alloca !quake.ref
    quake.h %0 : (!quake.ref) -> ()
    quake.x %0 : (!quake.ref) -> ()
    quake.y %0 : (!quake.ref) -> ()
    quake.z %0 : (!quake.ref) -> ()
    quake.t %0 : (!quake.ref) -> ()
    quake.s %0 : (!quake.ref) -> ()
    %measOut = quake.mz %0 name "" : (!quake.ref) -> !quake.measure
    return
  }
}
```

Reply:
Yep we can do that